helper/builtinplugins/builtin.go

```diff
@@ -0,0 +1,40 @@
+package builtinplugins
+
+import (
+	"github.com/hashicorp/vault/plugins/database/cassandra"
+	"github.com/hashicorp/vault/plugins/database/mssql"
+	"github.com/hashicorp/vault/plugins/database/mysql"
+	"github.com/hashicorp/vault/plugins/database/postgresql"
+)
+
+type BuiltinFactory func() (interface{}, error)
+
+var plugins map[string]BuiltinFactory = map[string]BuiltinFactory{
+	// These four plugins all use the same mysql implementation but with
+	// different username settings passed by the constructor.
+	"mysql-database-plugin":        mysql.New(mysql.DisplayNameLen, mysql.UsernameLen),
+	"aurora-database-plugin":       mysql.New(mysql.LegacyDisplayNameLen, mysql.LegacyUsernameLen),
+	"rds-database-plugin":          mysql.New(mysql.LegacyDisplayNameLen, mysql.LegacyUsernameLen),
```

Comment:
Interesting. It looks like that got added in November 2016, long after the original RFC for this. That's a good catch, and maybe we should just add some variants for those up-front, even if (like here) they are the same to start out with. @briankassouf what do you think?